### PR TITLE
DateSelect component style and props improvements

### PR DESCRIPTION
**Pull request by ogupte**
## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [x] IE11 (Win 7)
  - [x] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [ ] Two core team engineer approvals
- ~~One core team UX approval~~

### CODE REVIEW

File: src/components/DateSelect/DateSelect.jsx [hunk 1]
```diff
@@ -201,6 +216,30 @@ const DateSelect = createClass({
...
+			const rootElement = ReactDOM.findDOMNode(this.rootRef);
+			const {
+				width,
+				height,
+			} = rootElement.getBoundingClientRect();
+			const navButtonsWidth = NAV_BUTTON_SIZE * 2;
+			const oneMonthShownWidth = (width - navButtonsWidth) / monthsShown + navButtonsWidth;
+			const size = Math.sqrt(oneMonthShownWidth * height);
+			const relativeFontSize = Math.round(size / 24);
+			const relativeMinWidth = (((width - navButtonsWidth) / monthsShown) * 10.1075 / relativeFontSize) * monthsShown + navButtonsWidth;
```
Comment (jondlm) [inline src/components/DateSelect/DateSelect.jsx]: `10.1075`? Did it just look good in your testing?

File: src/components/DateSelect/DateSelect.jsx [hunk 2]
```diff
@@ -201,6 +216,30 @@ const DateSelect = createClass({
...
+
+			rootElement.style.fontSize = `${relativeFontSize}px`;
```
Comment (jondlm) [inline src/components/DateSelect/DateSelect.jsx]: So this'll only change once at the component mount, is that an issue for the prop changing?

File: src/components/DateSelect/DateSelect.jsx (hunk 1 above)
Comment (ogupte) [inline src/components/DateSelect/DateSelect.jsx]: yeah i calculated this as the ratio that the default font-size with the default size of the DateSelect. so in order to scale the font-size, i use the same ratio here

File: src/components/DateSelect/DateSelect.jsx (hunk 2 above)
Comment (ogupte) [inline src/components/DateSelect/DateSelect.jsx]: it's documented in the prop as only affecting the _initial_ font-size so i think it's ok. i'd rather not have to recompute onresize and on props change

Comment (jondlm) [inline src/components/DateSelect/DateSelect.jsx]: Cool, as long as we call it out I'm fine with that

**Pull request merged**